skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,125 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    def __init__(self, interval_metadata=None):
+        raise NotImplementedError
+
+    def _init_(self, interval_metadata=None):
```

Comment:
Please structure methods in this class like they are in the positional metadata mixin. I'm trying to keep the 3 mixin classes as consistent as possible so it's easier to maintain and reason about them, and certain design decisions were made for safety and performance. It'll also make writing the tests easier because you can model those off of `metadata` or `positional_metadata`. For example, `_init_` should be:

```python
def _init_(self, interval_metadata=None):
    if interval_metadata is None:
        # Could use deleter but this is less overhead and needs to be fast.
        self._interval_metadata = None
    else:
        # Use setter for validation and copy.
        self.interval_metadata = interval_metadata
```

I'll mark any other methods that need updating to match positional metadata mixin.